src/controllers/restatedeployment/reconcilers/knative.rs

```diff
@@ -0,0 +1,938 @@
+
+    // Build Configuration spec
+    let config_spec = build_configuration_spec(rsd, &config_name, namespace, tag)?;
+
+    info!(
+        configuration_name = %config_name,
+        namespace = %namespace,
+        tag = %tag,
+        "Applying Knative Configuration"
+    );
+
+    // Apply Configuration using server-side apply
+    let config_api: Api<Configuration> = Api::namespaced(ctx.client.clone(), namespace);
+    let params = PatchParams::apply("restate-operator").force();
+
+    let config = config_api
+        .patch(&config_name, &params, &Patch::Apply(&config_spec))
+        .await?;
+
+    Ok(config)
+}
+
+/// Build Configuration resource specification
+fn build_configuration_spec(
+    rsd: &RestateDeployment,
+    name: &str,
+    namespace: &str,
+    _tag: &str,
+) -> Result<serde_json::Value> {
+    let knative_spec = rsd
+        .spec
+        .knative
+        .as_ref()
+        .ok_or_else(|| Error::InvalidRestateDeployment("Missing knative spec".into()))?;
+
+    // Build revision template annotations
+    let mut annotations = BTreeMap::new();
+
+    // Step 1: Start with user-provided revision annotations
+    if let Some(user_annotations) = &knative_spec.revision_annotations {
+        annotations.extend(user_annotations.clone());
+    }
+
+    // Step 2: Apply operator-managed autoscaling annotations (overriding user values)
+    if let Some(min) = knative_spec.min_scale {
+        annotations.insert("autoscaling.knative.dev/min-scale".to_string(), min.to_string());
+    } else {
+        // Default to scale-to-zero
+        annotations.insert("autoscaling.knative.dev/min-scale".to_string(), "0".to_string());
+    }
+
+    if let Some(max) = knative_spec.max_scale {
+        annotations.insert("autoscaling.knative.dev/max-scale".to_string(), max.to_string());
+    }
+
+    if let Some(target) = knative_spec.target {
+        annotations.insert("autoscaling.knative.dev/target".to_string(), target.to_string());
+    }
+
+    // Step 3: Always set parent deployment tracking (operator-managed)
+    annotations.insert(
+        RESTATE_DEPLOYMENT_ANNOTATION.to_string(),
+        rsd.name_any(),
+    );
+
+    // Get container spec from template
+    let containers = rsd
+        .spec
+        .template
+        .spec
+        .as_ref()
+        .and_then(|spec| spec.get("containers"))
+        .ok_or_else(|| Error::InvalidRestateDeployment("Missing containers in template".into()))?;
+
+    // Ensure Restate port and validate for Knative compatibility
+    let mut containers_array: Vec<serde_json::Value> = serde_json::from_value(containers.clone())?;
+    validate_knative_containers(&containers_array)?;
+    ensure_restate_port(&mut containers_array)?;
+    ensure_readiness_probe(&mut containers_array)?;
+
+    // Create owner reference
+    let owner_reference = rsd.controller_owner_ref(&()).unwrap();
+
+    Ok(json!({
+        "apiVersion": "serving.knative.dev/v1",
+        "kind": "Configuration",
```

Comment:
for replicaset i used `DynamicObject` instead of json::Value, then at least we can have typed values for the kind and the metadata and just use serde json for the spec